Assets/Scripts/Algorithms/Patrolling/CognitiveCoordinated/CognitiveCoordinatedBase.cs

```diff
@@ -34,25 +33,11 @@ protected override void GetDebugInfo(StringBuilder stringBuilder)
         protected override IComponent[] CreateComponents(IRobotController controller, PatrollingMap patrollingMap)
         {
             _controller = controller;
-            _goToNextVertexComponent = new GoToNextVertexComponent(NextVertex, this, controller, Coordinator.GlobalMap);
+            _goToNextVertexComponent = new GoToNextVertexComponent(NextVertex, this, controller, patrollingMap);
```

Comment:
Are you sure you want this? The global one should still use the global map.

Reply:
It is used to get the initial vertex. I think you should make it use the global map. It is not smart to mix vertices from different maps.

Reply:
It's fine 💯

Reply:
The maps should be exactly the same, the only difference being, the patrolling map are copies without references to each other, such that robots cant smuggle info, right?